scripts/SREP/etcd-complete-health-check/script.sh

```diff
@@ -0,0 +1,57 @@
+#!/bin/bash
+
+echo "[INFO] Complete ETCD Status"
+
+## Get the ETCD pod
+ETCD_POD=$(oc get pods -l k8s-app=etcd -n openshift-etcd -o jsonpath='{.items[*].metadata.name}' | awk '{ print $1 }')
+
+## Print out the member table
+echo
+echo "[INFO] etcdctl member list"
+oc exec "$ETCD_POD" -n openshift-etcd -c etcdctl -- sh -c "etcdctl member list -w table"
+
+## Print out the summary of the etcd members
+echo
+echo "[INFO] summary of etcd"
+ETCD_SUMMARY=$(oc get etcd -o=jsonpath='{range .items[0].status.conditions[?(@.type=="EtcdMembersAvailable")]}{.message}{"\n"}')
+echo "$ETCD_SUMMARY"
+
+## Print out the summary of Endpoint Status
+echo
+echo "[INFO] etcdctl endpoint status"
+endpoint_status=$(oc exec "$ETCD_POD" -n openshift-etcd -c etcdctl -- sh -c "etcdctl endpoint status -w table" 2>/dev/null)
+echo "$endpoint_status" 
+
+
+## Print out the summary of Endpoint Health
+echo
+echo "[INFO] etcdctl endpoint health"
+endpoint_health=$(oc exec "$ETCD_POD" -n openshift-etcd -c etcdctl -- sh -c "etcdctl endpoint health -w table" 2>/dev/null)
+echo "$endpoint_health"
+
+
+# Get the logs for the unhealthy node(s)
+echo "$ETCD_SUMMARY" | sed -n 1'p' | tr ',' '\n' | grep unhealthy  |  while read word; do
+    name=$(echo "$word" | awk '{ print $1 }')
+    echo
+    echo "[INFO] logs for $name"
+    echo   
+    if command=$(oc logs -c etcd -n openshift-etcd etcd-"$name")
+    then
+        echo "$command"
+    elif cmd=$(oc get nodes | grep master | awk '{print $1}' | sed 1q |
```

Comment:
If we're just wanting to get a list of master nodes, something like 

```
oc get nodes -l node-role.kubernetes.io/master="" -o name
```

might be better.